yargs.js

```diff
@@ -1010,7 +1018,14 @@ function Yargs (processArgs, cwd, parentRequire) {
     args = args || processArgs
 
     options.__ = y18n.__
-    options.configuration = pkgUp()['yargs'] || {}
+    options.configuration = self.getParserConfiguration()
+
+    // Deprecated
+    let pkgConfig = pkgUp()['yargs']
+    if (pkgConfig) {
+      console.warn('Configuring yargs through package.json is deprecated and will be removed in the next major release, please use the JS API instead.')
```

Comment:
There's no other "raw" `console.*` call in this file, we should probably use the instance-scoped [`_logger.error()`](https://github.com/yargs/yargs/blob/27bf73923423dbe84dd2fd282fdd31d26bdb6cee/yargs.js#L956-L975) here instead.

Parenthetically, `console.warn` vs `console.error` is just semantics, it doesn't call `process.exit()` and both console methods log to `process.stderr` anyway.

Reply:
Alright, thanks for reviewing the PR ! 41fe766

Reply:
Using the scoped logger happens to break the tests as this deprecation warning is unexpected. I've reverted the commit, I'm not sure it's relevant to pass this warning for non-cli print uses.

Reply:
Okay, fair enough. A pity about the brittle tests.